src/gol.rs

```
@@ -0,0 +1,314 @@
+use macroquad::prelude::*;
+use std::{cmp::min, collections::HashSet};
+
+/// Implementation of game of life. The struct has two field for the boards dimentions and anotherone to register
+/// the living cells. After each transition this set is updated with the resulting living cells.
+pub struct GameOfLife {
+    width: usize,
+    height: usize,
+    alive_cells: HashSet<(usize, usize)>,
+}
+
+impl GameOfLife {
+    /// Creates an instance of a Game of Life with an empty set of alive cells and the dimentions of the board.
+    pub fn new(width: usize, height: usize) -> Self {
+        let alive_cells = HashSet::new();
+        Self {
+            width,
+            height,
+            alive_cells,
+        }
+    }
+    /// Adds a cell to the alive cells set.
+    /// TODO: handle coordinates outiside the boards range with error
```

This is not optional, silent failure is not an option.

---

Roger that! Fixing it right away!